pysnooper/tracer.py

```diff
@@ -212,6 +217,7 @@ def __init__(self, output=None, watch=(), watch_explode=(), depth=1,
              for v in utils.ensure_tuple(watch_explode)
         ]
         self.frame_to_local_reprs = {}
+        self.start_times = []
```

Comment:
A stack is not good enough, this will not handle multiple threads. As I said before, you need a mapping from frames to times, similar to `frame_to_local_reprs`.

Reply:
Alex is probably right here.